src/tactics/FStar.Tactics.V2.Basic.fst

```diff
@@ -2202,18 +2260,21 @@ let refl_core_compute_term_type (g:env) (e:term) (eff:tot_or_ghost) : tac (optio
          dbg_refl g (fun _ ->
            BU.format1 "refl_core_compute_term_type: %s\n" (Print.term_to_string e));
          let must_tot = to_must_tot eff in
+         let guards : ref (list (env & typ)) = BU.mk_ref [] in
          let gh = fun g guard ->
-           Rel.force_trivial_guard g
-             {Env.trivial_guard with guard_f = NonTrivial guard};
-           true in
+           (* FIXME: this is kinda ugly, we store all the guards
```

Comment:
Yeah ... the callback could be `proc_guard_formula` here and then the meta function could return `[]` ... though that is not uniform with other callbacks.

Reply:
I think we can't do that since these must be pure (non-`tac`) functions, both the callback that the core typechecker receives and the function wrapped by `refl_typing_builtin_wrapper`--- but `proc_guard_formula` must be in `tac`.